webapp/src/onboarding.ts

```diff
@@ -0,0 +1,76 @@
+export interface querySelector {
+    targetQuery: string;
+    sansQuery?: string;
+    sansLocation?: pxt.tour.BubbleLocation;
+}
+
+function getTargetMap(target: string): querySelector {
+    const targetMap: pxt.Map<querySelector> = {
+        "simulator": {
+            targetQuery: "#boardview",
+        },
+        "toolbox": {
+            targetQuery: ".blocklyToolboxDiv",
+        },
+        "workspace": {
+            targetQuery: "#blocksEditor", // includes the toolbox
+            sansQuery: ".blocklyToolboxDiv",
+            sansLocation: pxt.tour.BubbleLocation.Left
+        },
+        "share": {
+            targetQuery: ".shareproject",
+        },
+        "download": {
+            targetQuery: "#downloadArea",
+        },
+        "everything" : {
+            targetQuery: "#root",
+        }
+    }
+    return targetMap[target];
+}
+
+export async function loadTourStepsAsync(name: string): Promise<pxt.MarkdownSection[]> {
+    return pxt.Cloud.markdownAsync(name)
+        .then(md => pxt.getSectionsFromMarkdownMetadata(md));
```

Comment:
```suggestion
export async function loadTourStepsAsync(name: string): Promise<pxt.MarkdownSection[]> {
    const md = pxt.Cloud.markdownAsync(name);
    return pxt.getSectionsFromMarkdownMetadata(md);
```

prefer async/await

Reply:
Is `await` not needed on line 34?

Reply:
whoops, I got distracted in a call while typing this comment, it is definitely needed! edited